src/test/java/bio/terra/integration/DataRepoClient.java

```diff
@@ -105,7 +105,18 @@ public <T> DataRepoResponse<T> put(
 
   public <T> DataRepoResponse<T> delete(
       TestConfiguration.User user, String path, TypeReference<T> responseClass) throws Exception {
-    HttpEntity<String> entity = new HttpEntity<>(getHeaders(user));
+    return delete(user, path, null, responseClass);
+  }
+
+  public <T> DataRepoResponse<T> delete(
+      TestConfiguration.User user, String path, String json, TypeReference<T> responseClass)
+      throws Exception {
+    HttpEntity<String> entity;
+    if (json != null) {
+      entity = new HttpEntity<>(json, getHeaders(user));
+    } else {
+      entity = new HttpEntity<>(getHeaders(user));
+    }
```

Comment:
The first argument to `HttpEntity(body, headers)` is nullable so you don't need an `if` here.
```suggestion
    HttpEntity<String> entity = new HttpEntity<>(json, getHeaders(user));
```